testing/citest/spinnaker_testing/spinnaker_test_scenario.py

```diff
@@ -444,57 +488,68 @@ def _do_init_bindings(self):
       if not bindings.get('TEST_AWS_VPC_ID', ''):
         # We need to figure out a specific aws vpc id to use.
-        logger = logging.getLogger(__name__)
+        logger=logging.getLogger(__name__)
         logger.info('Determine default AWS VpcId...')
-        vpc_list = self.__aws_observer.get_resource_list(
+        vpc_list=self.__aws_observer.get_resource_list(
             context,
-            root_key='Vpcs',
-            aws_command='describe-vpcs',
-            args=['--filters', 'Name=tag:Name,Values=defaultvpc'],
-            region=bindings['TEST_AWS_REGION'],
-            aws_module='ec2', profile=self.__aws_observer.profile)
+            root_key = 'Vpcs',
+            aws_command = 'describe-vpcs',
+            args = ['--filters', 'Name=tag:Name,Values=defaultvpc'],
+            region = bindings['TEST_AWS_REGION'],
+            aws_module = 'ec2', profile = self.__aws_observer.profile)
         if not vpc_list:
           raise ValueError('There is no vpc tagged as "defaultvpc"')
-        bindings['TEST_AWS_VPC_ID'] = vpc_list[0]['VpcId']
+        bindings['TEST_AWS_VPC_ID']=vpc_list[0]['VpcId']
         logger.info('Using discovered default VpcId=%s',
                     str(bindings['TEST_AWS_VPC_ID']))
 
       if not bindings.get('TEST_AWS_SECURITY_GROUP', ''):
         # We need to figure out a specific security group that is compatable
         # with the VpcId we are using.
-        logger = logging.getLogger(__name__)
+        logger=logging.getLogger(__name__)
         logger.info('Determine default AWS SecurityGroupId...')
-        sg_list = self.__aws_observer.get_resource_list(
+        sg_list=self.__aws_observer.get_resource_list(
             context,
-            root_key='SecurityGroups',
-            aws_command='describe-security-groups', args=[],
-            region=bindings['TEST_AWS_REGION'],
-            aws_module='ec2', profile=self.__aws_observer.profile)
+            root_key = 'SecurityGroups',
+            aws_command = 'describe-security-groups', args = [],
+            region = bindings['TEST_AWS_REGION'],
+            aws_module = 'ec2', profile = self.__aws_observer.profile)
         for entry in sg_list:
           if entry.get('VpcId', None) == bindings['TEST_AWS_VPC_ID']:
-            bindings['TEST_AWS_SECURITY_GROUP_ID'] = entry['GroupId']
+            bindings['TEST_AWS_SECURITY_GROUP_ID']=entry['GroupId']
             break
         logger.info('Using discovered default SecurityGroupId=%s',
                     str(bindings['TEST_AWS_SECURITY_GROUP_ID']))
     else:
-      self.__aws_observer = None
-      logger = logging.getLogger(__name__)
+      self.__aws_observer=None
+      logger=logging.getLogger(__name__)
       logger.warning(
           '--aws_profile was not set.'
           ' Therefore, we will not be able to observe Amazon Web Services.')
 
+
+  def __init_azure_bindings(self):
+    bindings = self.bindings  # base class made a copy
```

Comment:
In head, the observer creation will be in AzureScenarioSupport._make_observer(self)
The bindings initialization (e.g. TEST_AZURE_LOCATION) would go into AzureScenarioSupport.__init__(self, scenario) where you would update scenario.bindings.